src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -68,18 +75,28 @@ def acr_connected_registry_create(cmd,  # pylint: disable=too-many-locals, too-m
 
     ErrorResponseException = cmd.get_models('ErrorResponseException')
     parent = None
+    mode = mode.capitalize()
     if parent_name:
         try:
             parent = acr_connected_registry_show(cmd, client, parent_name, registry_name, resource_group_name)
+            connected_registry_list = list(client.list(resource_group_name, registry_name))
+            family_tree, _ = _get_family_tree(connected_registry_list, None)
         except ErrorResponseException as ex:
             if ex.response.status_code == 404:
                 raise CLIError("The parent connected registry '{}' could not be found.".format(parent_name))
             raise CLIError(ex)
-        if parent.mode.lower() != ConnectedRegistryModes.REGISTRY.value and parent.mode.lower() != mode.lower():
+        if parent.mode != ConnectedRegistryModes.REGISTRY.value and parent.mode != mode:
             raise CLIError("Can't create the registry '{}' with mode '{}' ".format(connected_registry_name, mode) +
                            "when the connected registry parent '{}' mode is '{}'. ".format(parent_name, parent.mode) +
                            "For more information on connected registries " +
                            "please visit https://aka.ms/acr/connected-registry.")
+        msg = "Can't create the registry '{}'. The ancestor connected ".format(connected_registry_name) +\
```

Comment:
Does RP also validate and block the creation if parent is inactive? If yes, I try to understand why CLI also needs to verify it. Do you want to avoid updating the permissions if CLI can detect the inactive earlier?

Reply:
I think RP only validates the direct parent, not all ancestors. Also, the reason I validate in the CLI is so that we do not update the permissions and create/override existing resources when it's going to fail. 
I can remove the check. However, I think it'll be a common scenario when they try to create a connected registry and update all ancestors, and create a sync token and scope map. If they then don't create the connected registry or change the name, use different repos etc... This will contribute to having some unused permissions or resources. That's why I had the check